### PR TITLE
Reverse method in  MelkmanConvexHull

### PR DESCRIPTION
**Pull request by kerautret**
Addition of a reverse method in order to process points coming from front or back in reference to a digital contour.
Différent tests were given and the doc is also updated.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -7,6 +7,10 @@
    the width (vertical/horizontal or Euclidean) of a convex hull.
    (Bertrand Kerautret, [#1052](https://github.com/DGtal-team/DGtal/pull/1052))
 
+ - MelkmanConvexHull: addition of a reverse method in order to process points
```
Comment (dcoeurjo) [inline ChangeLog.md]: addition of -> new ?

File: ChangeLog.md [hunk 2]
```diff
@@ -7,6 +7,10 @@
...
+   coming from front or back in reference to a digital contour.
```
Comment (dcoeurjo) [inline ChangeLog.md]: "New method to allow point insertions and convex hull computation on both side of a point sequence " ?

File: src/DGtal/geometry/doc/moduleHull2D.dox [hunk 3]
```diff
@@ -177,6 +177,31 @@ the on-line feature, you may call the off-line procedure melkmanConvexHullAlgori
 
 @warning The behavior is undefined if the input points do not form a simple polygonal line. 
 
+\subsubsection subsectmoduleHull2D131 Constructing a Melkman convex hull from front and back.
```
Comment (dcoeurjo) [inline src/DGtal/geometry/doc/moduleHull2D.dox]: from both sides ?

File: src/DGtal/geometry/doc/moduleHull2D.dox [hunk 4]
```diff
@@ -177,6 +177,31 @@ the on-line feature, you may call the off-line procedure melkmanConvexHullAlgori
...
+If you want to construction a convex hull by considering the front and back of a contour you will need to use the @red reverse() method. For instance, to recover the convex hull of the point P0 of the following figure, you will need to do as follows:
```
Comment (dcoeurjo) [inline src/DGtal/geometry/doc/moduleHull2D.dox]: 80col please ;)

File: src/DGtal/geometry/tools/MelkmanConvexHull.h [hunk 5]
```diff
@@ -211,6 +219,10 @@ namespace DGtal
      * Used to define a default functor to allow default constructor
      **/
     Functor myDefaultFunctor;
+    /**
+     * first point used to reverse the convexhull container.
+     **/
+    TPoint myFirst;
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/MelkmanConvexHull.h]: TPoint -> Point

Comment (dcoeurjo) [inline src/DGtal/geometry/tools/MelkmanConvexHull.h]: myFirstPoint?

File: src/DGtal/geometry/tools/MelkmanConvexHull.ih [hunk 6]
```diff
@@ -190,6 +192,27 @@ DGtal::MelkmanConvexHull<TPoint, TOrientationFunctor>::clear()
   myContainer.clear();
 }
 
+// ----------------------------------------------------------------------------
+template <typename TPoint, typename TOrientationFunctor>
+inline
+void
+DGtal::MelkmanConvexHull<TPoint, TOrientationFunctor>::reverse()
+{
+  // if convexhull is reducted to a simgle segment non need to reverse anything.
```
Comment (dcoeurjo) [inline src/DGtal/geometry/tools/MelkmanConvexHull.ih]: reduced 
single segment, no need

File: tests/geometry/tools/testConvexHull2DThickness.cpp [hunk 7]
```diff
@@ -128,6 +126,17 @@ TEST_CASE( "Testing Rotating Caliper of ConvexHull2D (convex hull with floating
...
        REQUIRE( sE == Point(102.3, 52.3) );
      }
+   ch.reverse();
+   DGtal::functions::Hull2D::computeHullThickness(ch.begin(), ch.end(), 
+                                                  DGtal::functions::Hull2D::EuclideanThickness,
+                                                  pE, qE, sE);
+   
+   SECTION("Testing same antipodal points after hull reversing ")
+     {
+       REQUIRE( pE == Point(101.2, 48.2) );
```
Comment (dcoeurjo) [inline tests/geometry/tools/testConvexHull2DThickness.cpp]: use approx?

File: ChangeLog.md (hunk 2 above)
Comment (kerautret) [inline ChangeLog.md]: yes thanks better ;)

File: src/DGtal/geometry/doc/moduleHull2D.dox (hunk 3 above)
Comment (kerautret) [inline src/DGtal/geometry/doc/moduleHull2D.dox]: yes

File: src/DGtal/geometry/tools/MelkmanConvexHull.h (hunk 5 above)
Comment (kerautret) [inline src/DGtal/geometry/tools/MelkmanConvexHull.h]: agree ;)

File: src/DGtal/geometry/tools/MelkmanConvexHull.ih (hunk 6 above)
Comment (kerautret) [inline src/DGtal/geometry/tools/MelkmanConvexHull.ih]: oups

File: tests/geometry/tools/testConvexHull2DThickness.cpp (hunk 7 above)
Comment (kerautret) [inline tests/geometry/tools/testConvexHull2DThickness.cpp]: with Point ? looks not working

**Comment by kerautret** [comment]
thanks for the review, that done expect Approx but not sure that needed.

File: tests/geometry/tools/testConvexHull2DThickness.cpp (hunk 7 above)
Comment (dcoeurjo) [inline tests/geometry/tools/testConvexHull2DThickness.cpp]: yeah.. you're right..

**Comment by dcoeurjo** [comment]
ok thanks.
merging

**Pull request merged**